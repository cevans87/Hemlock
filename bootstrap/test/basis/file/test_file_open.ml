open Basis

external user_data_pp: File.Open.t -> unit = "hm_basis_file_user_data_pp"
external sqring_pp: unit -> unit = "hm_basis_file_sqring_pp"

let () =
  let s = File.Open.submit ~flag:File.Flag.RW (Bytes.Slice.of_string_slice
      (String.C.Slice.of_string "file")) in
  user_data_pp s;
  sqring_pp ();
  let _ = File.Open.complete_hlt s in
  user_data_pp s;
  sqring_pp ();
  ()