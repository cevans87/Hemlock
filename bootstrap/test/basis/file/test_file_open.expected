user_data:
  refcount: 2
  pathname: "file"
  opcode:   IORING_OP_OPENAT
  cqe:
    res:       0
    flags:     0
    user_data: NULL
sqring:
  head:         0
  tail:         1
  ring_entries: 32
  ring_mask:    31
  flags:        0
  array: [0 ... 31]
  sqes:
    sqe:
      flags:    0
      ioprio:   0
      fd:       -100
      opcode:   IORING_OP_OPENAT
      user_data:
        refcount: 2
        pathname: "file"
        opcode:   IORING_OP_OPENAT
        cqe:
          res:       0
          flags:     0
          user_data: NULL
user_data:
  refcount: 1
  pathname: NULL
  opcode:   IORING_OP_OPENAT
  cqe:
    res:       4
    flags:     0
    user_data: <same as parent>
sqring:
  head:         1
  tail:         1
  ring_entries: 32
  ring_mask:    31
  flags:        0
  array: [0 ... 31]
  sqes:
